src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -110,17 +111,19 @@ private static IEnumerable<DValue<RecordValue>> LazyAddColumns(EvalVisitor runne
 
                     foreach (var column in newColumns)
                     {
-                        var value = column.Lambda.EvalAsync(runner, childContext).Result;
+                        var value = await column.Lambda.EvalAsync(runner, childContext);
                         fields.Add(new NamedValue(column.Name, value));
```

Comment:
var fields = new NamedValue[row.Value.Fields.Length + newColumns.Length];
.. copy row.Value.Fields by position to fields
here fields[i] = new NamedValue(...)
below remove .ToArray() below